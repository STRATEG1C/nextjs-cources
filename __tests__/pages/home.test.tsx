import '@testing-library/jest-dom';
import { render, screen } from '@testing-library/react';
import Page from '../../src/app/page';

describe('Home page', () => {
  it('renders page unchanged', () => {
    const { container } = render(<Page />);

    expect(container).toMatchSnapshot();
  });

  it('renders a heading and description', () => {
    render(<Page />);

    const heading = screen.getByRole('heading', {
      level: 1,
      name: "Hello, I'm your Next.js App",
    });

    const description = screen.getByText('Make web great again!');

    expect(heading).toBeInTheDocument();
    expect(description).toBeInTheDocument();
  });
});
