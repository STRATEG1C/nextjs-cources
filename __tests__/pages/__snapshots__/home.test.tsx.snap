// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home page renders page unchanged 1`] = `
<div>
  <main>
    <h1>
      Hello, I'm your Next.js App
    </h1>
    <p>
      Make web great again!
    </p>
  </main>
</div>
`;
